### ISSUE TITLE
Download presentation experience

### ISSUE THREAD

**Issue by tylercopeland** [bigbluebutton/bigbluebutton#4882]
This experience includes:
- download the entire presentation (without annotations)
- download the current slide (with annotation)
- download the entire presentation (with annotations)

**Comment by santoshitutor** [comment]
Can we add APIs as well to fetch these files with annotations after the meeting ends? We can have config setup to allow auto saving of the entire annotated presentation which can be accessed/downloaded through APIs.

**Comment by ffdixon** [comment]
That's the plan.  Your requests were related to APIs ... are you a developer?  This is on our road map, but it would be accelerated with help from others.

**Comment by ann0see** [comment]
Yep. Also adding a feature to download the ether pad shared notes especially in break out rooms would be great. Sometimes we had the problem to save shared notes fast enough (by copy and paste) before the breakout room closed.

**Comment by justAnormalUser00** [comment]
@ann0see have you tried bbb-playback-notes ? if you record the session the notes (ether pad) get saved automatically.

**Comment by ann0see** [comment]
No not yet. We don’t have recording enabled because of the GDPR.

 

Von: justAnormalUser00 <notifications@github.com> 
Gesendet: Donnerstag, 30. April 2020 10:17
An: bigbluebutton/bigbluebutton <bigbluebutton@noreply.github.com>
Cc: ann0see <jonathansmithchung@t-online.de>; Mention <mention@noreply.github.com>
Betreff: Re: [bigbluebutton/bigbluebutton] Add ability to download presentation with whiteboard annotations (#4882)

 

@ann0see <https://github.com/ann0see>  have you tried bbb-playback-notes ? if you record the session the notes (ether pad) get saved automatically.

**Comment by ann0see** [comment]
No we don't have enabled it due to the GDPR

**Comment by ayodeji** [comment]
Are there plans to still add this feature? I have a client asking and i'm wondering.

**Comment by mcferdev** [comment]
> Are there plans to still add this feature? I have a client asking and i'm wondering.

Yeah. Me too ! I thought it would be possible, but I just can download the original, but in my case I need a signature from the user. In my case I am thinking about incorporate in a mobile app and put the feature to sign, or the user download the original, sign and than, using a scan app, send back. But, too many steps...

**Comment by vol-and** [comment]
Are there any steps to implement this feature?

**Comment by GhaziTriki** [comment]
@vol-and We started initial architectural discussions regarding this feature few months ago. We will keep this issue updated if we have any news regarding it.

**Comment by irkreja** [comment]
More than 3 year passed but still no update regarding this issue. May be you will never do it. Just making false promises.

**Comment by prlanzarin** [comment]
> More than 3 year passed but still no update regarding this issue. May be you will never do it. Just making false promises.

Pull requests are welcome :)

**Comment by ajithgopi** [comment]
I've added some more details in #12125 which is now closed as it's a duplicate of this issue.

**Comment by danielpetri1** [comment]
I wrote [this script ](https://github.com/danielpetri1/BBB-Whiteboard/blob/main/export_annotated_slides.rb) to get the annotated slides as individual images from a recording's shapes.svg file.

With @ajithgopi 's suggestion of using  svg-to-pdfkit, this step of rendering PNGs can be probably skipped.

Feedback is more than welcome =)

**Comment by ajithgopi** [comment]
> I wrote [this script ](https://github.com/danielpetri1/BBB-Whiteboard/blob/main/export_annotated_slides.rb) to get the annotated slides as individual images from a recording's shapes.svg file.
> 
> With @ajithgopi 's suggestion of using  svg-to-pdfkit, this step of rendering PNGs can be probably skipped.
> 
> Feedback is more than welcome =)

But the recording should be enabled for that to work right? For large scale deployments, enabling recording unnecessarily can take up too much space. And we'll also need to wait for the recording to process before it can be downloaded. I was suggesting a method which allows the students to download the slides, with annotations, during the class... :)

**Comment by hiroshisuga** [comment]
Not necesarily, @ajithgopi . What takes time is movie encoding. Annotations and slides are converted almost instantly. It is possible, although currently not possible, to deploy the script so that BBB generates the annotated PDF without taking too much time.

**Comment by danielpetri1** [comment]
[This script](https://github.com/danielpetri1/bbb-recording-exporter/tree/pdf-export-server) will convert the slides of a BBB recording automatically into an annotated PDF once the meeting ends. There's a client-side version on the other branch.

**Comment by antobinary** [comment]
> download the current slide (with annotation)

This part should land in 2.5 via https://github.com/bigbluebutton/bigbluebutton/pull/14406

**Comment by knarrff** [comment]
If I understoof #14406 correctly, this only handles png. Information in this ticket here suggests that the annotations are actually already present in svg. Could we (also) be able to download just that?

**Comment by antobinary** [comment]
Implemented as part of 2.6 :confetti_ball: Nice work, @danielpetri1 :tada:

**Comment by piszczu4** [comment]
Is it possible to call it from bbb server once meeting ends?

**Comment by antobinary** [comment]
> Is it possible to call it from bbb server once meeting ends?

Not currently. As discussed in the community call today, your ideas for exporting various items (All presentations + annotations, full chat history,...) are neat. While not on the current roadmap for the dev team, they may gather steam with support from others. Would you like to open a new github enhancement issue describing the solution you'd like to see (and which portion of it you think you may be able to carry out)?

**Comment by GeCol1** [comment]
The module from @danielpetri1 was awesome but not available in >3.x versions of BBB.
We have this module https://codeberg.org/solsys-fr/bbb-pres-export instead that generates the annotated_slides.pdf file and update the metadata to appear in greenlight.
